sklearn_numba_dpex/common/kernels.py

```diff
@@ -633,6 +700,9 @@ def _make_partial_sum_reduction_2d_axis0_kernel(
     reduction_block_size = 2 * n_sub_groups_per_work_group
     work_group_shape = (n_sub_groups_per_work_group, sub_group_size)
 
+    local_sum_and_set_items_if = _make_sum_and_set_items_if_kernel_func()
+    global_sum_and_set_items_if = _make_sum_and_set_items_if_kernel_func()
```

Comment:
Why do we need two of those and not just one? Is it for this reason?

```suggestion
    # HACK: must define twice to work around the bug highlighted in
    # test_regression_fix
    local_sum_and_set_items_if = _make_sum_and_set_items_if_kernel_func()
    global_sum_and_set_items_if = _make_sum_and_set_items_if_kernel_func()
```

Reply:
No need to worry too much about this, the hack will be removed everywhere after bump to `0.20.0dev3`, the PR has been opened https://github.com/soda-inria/sklearn-numba-dpex/pull/93, and hope can be merged soon after this one